src/odemis/gui/cont/microscope.py

```diff
@@ -1441,3 +1441,84 @@ def _update_load_time(self, future, start, end):
         rem_time = est_end - time.time()
         logging.debug("Loading future remaining time: %f", rem_time)
 
+
+class FastEMStateController(object):
+    """
+    Manages the chamber pressure and ebeam power states.
+    """
+
+    def __init__(self, tab_data, tab_panel):
+        """ Binds the 'hardware' buttons (pressure and ebeam) to their appropriate
+        Vigilant Attributes in the tab and GUI models
+
+        tab_data (MicroscopyGUIData): ebeam, chamber
+        tab_panel: (wx.Panel): the microscope tab
+        """
+        self._main_data = tab_data.main
+        self._tab_data = tab_data
+        self._tab_panel = tab_panel
+
+        # E-beam management
+        tooltips = {0: "Truen E-beam on", 1: "Turn E-beam off"}
+        self._ebeam_btn_ctrl = HardwareButtonController(tab_panel.btn_ebeam,
+                                                        self._main_data.ebeam_state,
+                                                        tooltips)
+        self._main_data.ebeam_state.subscribe(self._on_ebeam_state)
+        self._main_data.ebeam.power.subscribe(self._on_ebeam_power)
```

Comment:
Add `init=True`, so that it's synchronized at the beginning based on the current state of the ebeam.